### ISSUE TITLE
${parameter=word} not recognized as assignment

### ISSUE THREAD

**Issue by girst** [koalaman/shellcheck#1758]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2223
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

`${parameter=word}` is very similar `${parameter:=word}`, except that it substitutes parameter even if it is defined *null*. Both are [specified by POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_06_02). Shellcheck recognizes the latter as an assignment, but not the former and issues a confusing warning.

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

: "${foo:=foo}"  # works
: "${bar=bar}"  # warns "SC2154: bar is referenced but not assigned."

echo "$foo $bar"
```

#### Here's what shellcheck currently says:

> SC2154: bar is referenced but not assigned.

#### Here's what I wanted or expected to see:

nothing
